### DISCUSSION TITLE
Support for Challenger RP2040 boards?

### DISCUSSION THREAD

**Discussion by csku25** [micropython/micropython#15725] [RP2040 / Pico]
Hey, before I flash anything to my newest purchase, the [Challenger RP2040 WiFi/BLE MkII with chip antenna and 16bit accelerometer,](https://ilabs.se/product/challenger-rp2040-wifi-ble-mkii-with-chip-antenna-and-16bit-accelerometer/) I was wondering if there was a version of micropython for this board that would allow the rp2040 to communicate with the onboard ESP32-C3FN4 from Espressif and onboard MC3419 accelerometer. **I know about the circuitpython version by adafruit.** Please let me know where I can start!

**Comment by peterhinch** [comment]
The standard RP2040 firmware should work. The ESP with AT firmware will be connected to a UART - you'll need to find out which one. The MC3419 may be more of a problem. There is [this driver](https://github.com/jposada202020/Micropython_MC3479/tree/master) for the MC3479. You'll need to check how similar this is, and find out which I2C interface is used to communicate with it.
